File: SecAppBack/SecApp.Api/Models/Agent.cs

using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace SecApp.Api.Models
{
    public class Agent
    {
        public int AgentId { get; set; }
        [Required(ErrorMessage ="El campo {0} es requerido")]
        [StringLength(50, ErrorMessage ="El campo {0} debe tener {1} caracteres o menos")]
        public required string Name { get; set; } = string.Empty;
        [Required]
        [StringLength(50, ErrorMessage = "El campo {0} debe tener {1} caracteres o menos")]
        public string LastName { get; set; } = string.Empty;
        [Required]
        [StringLength(10, ErrorMessage ="Campo {0}: debe tener {1} digitos")]
        public string Phone { get; set; } = string.Empty;
        [Required]
        [StringLength(11, ErrorMessage = "El campo {0} debe tener {1} caracteres o menos")]
        public string Identification { get; set; } = string.Empty;
        public string? Email { get; set; } 
        [Required]
        public DateTime BirthDay { get; set; }
        [Required]
        public bool Status { get; set; }
        [Unicode(false)]
        public string? Photo { get; set; } 
        [Required]
        public int AgentCode { get; set; }
        public int RangeId { get; set; }
        public Range? Range { get; set; } // Navigation property to Range entity
        public int InstitutionId { get; set; }
        public Institution? Institution { get; set; } // Navigation property to Institution entity

    }
}

File: SecAppFront/src/app/agents/models/agent.ts
export interface AgentDto {
  agentId: number | null;
  name: string | null;
  lastName: string | null;
  phone: string | null;
  identification: string | null;
  email?: string;
  birthday: Date | null;
  status: boolean | null;
  photo?: string;
  agentCode: number | null;
  rangeId: number | null;
  rangeName: string | null;
}

export interface AgentCreateDto {
  //agentId: number | null;
  name: string;
  lastName: string;
  phone: string;
  identification: string;
  email?: string;
  birthday: Date;
  status: boolean;
  photo?: File;
  rangeId: number;
  institutionId: number;
  agentCode: number;
}

export interface AgentAutoCompleDto {
  id: number;
  name: string;
  lastName: string;
  phone: string;
  photo: string;
}
